### ISSUE TITLE
Has the inference been speed-up already?

### ISSUE THREAD

**Issue by Echo0117** [UKPLab/sentence-transformers#628]
Hi, I'd like to ask when using `model.encode` (the code is shown below), if the inference process is speed-up？If it is, is it with tensorFT or other methods? 

Also, what is the approximate max QPS when we are doing inference on the production environment？Thank you in advance!

```
model = SentenceTransformer('xlm-r-distilroberta-base-paraphrase-v1')
sentence_embeddings = model.encode(text)
```

**Comment by nreimers** [comment]
Hi @Echo0117 
The computation is based on Pytorch. There are some optimization with respect to batching the sentences, so that minimal padding is needed. Otherwise, there are no further optimizations applied by us (but of course, Pytorch and maybe Huggingface Transformers apply independently some optimizations).

Here you can find some speed measurements on a V100 GPU:
https://docs.google.com/spreadsheets/d/14QplCdTCDwEmTqrn1LH4yrbKvdogK4oQvYO1K1aPR5M/edit#gid=0

Note, speed largely depends on the length of your sentences. Transformer models have quadratic runtime with the sentence length, i.e., if your sentence is twice as long, speed will be 4 times slower.

Speed measurements were done on the SNLI+MultiNLI dataset.

**Comment by EquatorSavage** [comment]
Did you set this?
```
import torch
torch.set_num_threads(64)
```

**Comment by nreimers** [comment]
Set_num_threads is not called.

**Comment by EquatorSavage** [comment]
It can improve CPU utilization when QPS is high.

**Comment by Echo0117** [comment]
Hi Nils,

Thanks for your reply. Do you know what kind of optimizations has done by
Pytorch or Huggingface Transformers? Because I wanna speed up the Inference
 using TensorRT, do you have any ideas on how to do that? I have tried
different batch_sizt, but it didn't improve much ( my sentences are usually
less than 100 words).
The model I am using is 'xlm-r-distilroberta-base-paraphrase-v1'. Thank you
very much!

Nils Reimers <notifications@github.com> 于2020年12月16日周三 下午6:08写道：

**Comment by Echo0117** [comment]
Thanks for all your reply :)  I have tried different batch_size, but it didn't improve much (my sentences are usually
less than 100 words).

I still got a few more questions:

1. Do you know what kind of optimizations has done by Pytorch or Huggingface Transformers? 
2. Also, I wanna speed up the Inference using TensorRT, do you have any ideas on how to do that?
3. I tried to add self.half() at the end of the def __init__, so it changed it to FP16, and it improved the speed a bit, but I am not sure if it's the correct way to do it considering the effect of sentence similarity, since there are slight changes in my embedding matrices.

The model I am using is 'xlm-r-distilroberta-base-paraphrase-v1'. Thank you very much!

**Comment by nreimers** [comment]
1) Pytorch will optimize it in various ways. What specifically you have to check pytorch code.

2) Never used TensorRT, cannot help.

3) Sounds right. The effect should be small on the embeddings.

Best
Nils

**Comment by Echo0117** [comment]
Thank you very much! That helps a lot, I will close the issue~

**Comment by yuanzhoulvpi2017** [comment]
i have the same problem to speed up encode of sentence transformer . so I convert sentence-transformer model to onnx model and tensorrt model. **about 4 times faster**   you can use my tutorial：
 [[quick_sentence_transformers](https://github.com/yuanzhoulvpi2017/quick_sentence_transformers)](https://github.com/yuanzhoulvpi2017/quick_sentence_transformers)

![image](https://user-images.githubusercontent.com/30610553/156150479-b6b8e907-4345-436c-a1ac-df2227221e69.png)

**Comment by datduyng** [comment]
> Did you set this?
> 
> ```
> import torch
> torch.set_num_threads(64)
> ```

This 10x my model performance. Thanks a lot 👏
